src/services/authentication/authentication.context.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, createContext, useEffect } from "react";
+import firebase from "firebase/app";
+
+import { loginRequest, passwordResetRequest } from "./authentication.services";
+
+export const AuthenticationContext = createContext();
+
+export const AuthenticationContextProvider = ({ children }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState(null);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const unlisten = firebase.auth().onAuthStateChanged((usr) => {
+      if (usr) {
```

Comment:
better replace `usr` to `user`. What do you think? That would easier for the readability.